scipy/stats/tests/test_variation.py

```diff
@@ -32,9 +34,13 @@ def test_sign(self, sgn, xp):
         expected = xp.asarray(sgn*math.sqrt(2)/3)
         xp_assert_close(v, expected, rtol=1e-10)
 
+    @xfail_xp_backends(np_only=True, reason="IndexError: tuple index out of range")
```

Comment:
Why not skip other backends?

Reply:
Changing to explicitly skip the other backends.
Are you suggesting there is value in explicitly testing an input that is an actual numpy scalar instead of a scalar numpy array?

Reply:
I didn't investigate the code, but the fact that array-api-strict fails too strongly hints at a bug in the scipy.stats code itself

Reply:
> Are you suggesting there is value in explicitly testing an input that is an actual pure-python scalar instead of a scalar numpy array?

I don't think there's value in the behavior to begin with. This is a reducing function. If the input is 0d and you reduce along `axis` 0 (which doesn't exist), what should happen?

I commented because I was just trying to understand the motivation for the change in what was being tested. We try to preserve existing behavior with Python floats, but I don't think it needs to extend to Array API 0d arrays, so it seems just a tiny bit better to leave the test with Python floats. 

> I didn't investigate the code, but the fact that array-api-strict fails too strongly hints at a bug in the scipy.stats code itself

Maybe take that up with NumPy and JAX first:
```python3
import numpy as np
np.mean(np.asarray(1.), axis=0)
# numpy.exceptions.AxisError: axis 0 is out of bounds for array of dimension 0

import jax.numpy as xp
xp.mean(xp.asarray(1.), axis=0)
# IndexError: tuple index out of range
```
LMK what they say. But I don't think reducing functions should _need_ to be able to reduce along an axis that doesn't exist. I wonder if it was intentional in `torch`.

```python3
import torch
torch.mean(torch.asarray(1.), axis=0)
# tensor(1.)
```

Reply:
>  I don't think reducing functions should need to be able to reduce along an axis that doesn't exist. 

Definitely they shouldn't

> Maybe take that up with NumPy and JAX first:

Why does the test pass with a 0d numpy array though?

Reply:
Anyway, I've reverted the test